.github/workflows/performance.yml

```diff
@@ -0,0 +1,160 @@
+  # - Starts the WordPress Docker container.
+  # - Logs the running Docker containers.
+  # - Logs Docker debug info.
+  # - Installs WordPress.
+  # - Installs the Server Timing MU plugin.
+  # - Install wordpress importer plugin and import dummy data.
+  # - Update permalink structure.
+  # - Run the Performance tests (current commit).
+  # - Print Performance tests results.
+  # - Update the environment to 6.1.1
+  # - Run Performance tests (6.1.1)
+  # - Print Performance tests results (6.1.1).
+  # - Set commit details.
+  # - Publish performance results.
+  # - Ensures version-controlled files are not modified or deleted.
+  performance:
+    runs-on: ubuntu-latest
+    if: ${{ github.repository == 'WordPress/wordpress-develop' || github.event_name == 'pull_request' }}
+
+    steps:
+      - name: Configure environment variables
+        run: |
+          echo "PHP_FPM_UID=$(id -u)" >> $GITHUB_ENV
+          echo "PHP_FPM_GID=$(id -g)" >> $GITHUB_ENV
+
+      - name: Checkout repository
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+
+      - name: Set up Node.js
+        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3.6.0
+        with:
+          node-version-file: '.nvmrc'
+          cache: npm
+
+      - name: Log debug information
+        run: |
+          npm --version
+          node --version
+          curl --version
+          git --version
+          svn --version
+          locale -a
+
+      - name: Install npm Dependencies
+        run: npm ci
+
+      - name: Build WordPress
+        run: npm run build
+
+      - name: Start Docker environment
+        run: |
+          npm run env:start
+
+      - name: Log running Docker containers
+        run: docker ps -a
+
+      - name: Docker debug information
+        run: |
+          docker -v
+          docker-compose -v
+          docker-compose run --rm mysql mysql --version
+          docker-compose run --rm php php --version
+          docker-compose run --rm php php -m
+          docker-compose run --rm php php -i
+          docker-compose run --rm php locale -a
+
+      - name: Install WordPress
+        run: npm run env:install
+
+      - name: Install WordPress Importer plugin and import mock data
+        run: |
+          npm run env:cli -- plugin install wordpress-importer --activate
+          curl -O https://raw.githubusercontent.com/WPTT/theme-test-data/b9752e0533a5acbb876951a8cbb5bcc69a56474c/themeunittestdata.wordpress.xml
+          npm run env:cli -- import themeunittestdata.wordpress.xml --authors=create
+          rm themeunittestdata.wordpress.xml
+
+      - name: Update permalink structure
+        run: |
+          npm run env:cli -- rewrite structure '/%year%/%monthnum%/%postname%/'
+
+      - name: Install MU plugin
+        run: |
+          mkdir ./build/wp-content/mu-plugins
+          cp ./tests/performance/mu-plugins/server-timing.php ./build/wp-content/mu-plugins/server-timing.php
```

Comment:
I'm not familiar with best practices for Jest testing setups. But my first thought was "should this `mu-plugins` folder be moved within a `data` folder similar to how our PHPUnit folder is structured to make it more clear that the `mu-plugins` directory is meant to be copied into the install before running tests?"

Reply:
I don't have a strong preference on naming conventions here. This folder/file only exists to be copied into the environment during the workflow. If it's clearer to move this into a `data` folder, I'm happy to do so.

Reply:
To elaborate on my thinking a bit. I was thinking what if a plugin or theme is added in the future. Then we'd have a `themes` and `plugins` directory in this main `tests/performance` folder as well. We could always make a decision if that should happen.

@felixarntz or @youknowriad any preference here? Don't feel strongly either way.

Reply:
How about we move this to `tests/performance/wp-content/mu-plugins`? `wp-content` as a containing folder would be a very clear name for everyone I think :)

Reply:
> How about we move this to `tests/performance/wp-content/mu-plugins`? `wp-content` as a containing folder would be a very clear name for everyone I think :)

Works for me!

Reply:
Handled in [62a288d](https://github.com/WordPress/wordpress-develop/pull/4139/commits/62a288d2c8b2222dfc3f471404de1c1cd2ed42b4)